File: MarkdownEditor/MarkdownEditor/Sources/Editor/EditorDocumentStore.m
#import "EditorDocumentStore.h"

@implementation EditorDocumentStore

- (NSString *)readStringFromURL:(NSURL *)url error:(NSError **)error {
    return [NSString stringWithContentsOfURL:url encoding:NSUTF8StringEncoding error:error];
}

- (BOOL)writeString:(NSString *)string toURL:(NSURL *)url error:(NSError **)error {
    return [string writeToURL:url atomically:YES encoding:NSUTF8StringEncoding error:error];
}

@end

File: MarkdownEditor/MarkdownEditor/Sources/Editor/EditorDocumentStore.swift
import Foundation

@objc(EditorDocumentStore)
class EditorDocumentStore: NSObject {
    @objc(readStringFromURL:error:)
    func readString(from url: URL) throws -> String {
        try String(contentsOf: url, encoding: .utf8)
    }

    @objc(writeString:toURL:error:)
    func writeString(_ string: String, to url: URL) throws {
        try string.write(to: url, atomically: true, encoding: .utf8)
    }
}
